cognite/client/data_classes/_base.py

```diff
@@ -503,7 +503,7 @@ def dump(self, camel_case: Literal[True] = True) -> dict[str, Any]:
 
     @classmethod
     @abstractmethod
-    def _get_update_properties(cls) -> list[PropertySpec]:
+    def _get_update_properties(cls, item: CogniteResource | None = None) -> list[PropertySpec]:
```

Comment:
This change is to allow the updated properties to be different depending on whether InstanceId is passed or not.